src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensEditsHandler.cs

```diff
@@ -35,16 +35,20 @@ public SemanticTokensEditsHandler(SemanticTokensCache tokensCache)
             _tokensCache = tokensCache;
         }
 
-        public TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.SemanticTokensEditsParams request) => request.TextDocument;
+        public TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.SemanticTokensEditsParams request)
+        {
+            Contract.ThrowIfNull(request.TextDocument);
```

Comment:
Is it worth creating an abstract base class for these handlers that does this contract check? Could be used for every handler that expects a document, potentially.

Reply:
Yes.  I think we should do this.  But in a followup PR.  We should make this a standard thing that all handlers have to participate in.  Let's discuss on monday the best way to do this.  I could see this being a property, constructor param, a different interface, etc.